hivemq-edge/src/frontend/src/extensions/datahub/components/forms/ResourceNameCreatableSelect.tsx

```diff
@@ -0,0 +1,194 @@
+  const chakraProps = getChakra({ uiSchema: props.uiSchema })
+  const [options, setOptions] = useState(defaultOptions)
+
+  useEffect(() => {
+    if (defaultOptions.length) {
+      const options = [...defaultOptions]
+      const { isDraft } = props.options
+      if (isDraft)
+        options.push({
+          name: props.value,
+          versions: [],
+        })
+      setOptions(options)
+    }
+  }, [defaultOptions, props.options, props.value])
+
+  const onCreatableSelectChange = useCallback<
+    (newValue: OnChangeValue<ResourceFamily, false>, actionMeta: ActionMeta<ResourceFamily>) => void
+  >(
+    (newValue, actionMeta) => {
+      if (actionMeta.action === 'select-option' && newValue) {
+        props.onChange(newValue.name)
+        return
+      }
+      if (actionMeta.action === 'create-option' && newValue) {
+        props.onChange(newValue.label)
+        return
+      }
+    },
+    [props]
+  )
+
+  const handleCreate = (inputValue: string) => {
+    setOptions((prev) => [...(prev || []), createNewOption(inputValue)])
+    props.onChange(inputValue)
+  }
+
+  const value = options?.find((e) => e.name === props.value)
+  return (
+    <FormControl
+      {...chakraProps}
+      isDisabled={props.disabled || props.readonly}
+      isRequired={props.required}
+      isReadOnly={props.readonly}
+      isInvalid={props.rawErrors && props.rawErrors.length > 0}
+    >
+      {labelValue(
+        <FormLabel htmlFor={props.id} id={`${props.id}-label`}>
+          {props.label}
+        </FormLabel>,
+        props.hideLabel || !props.label
+      )}
+
+      <CreatableSelect<ResourceFamily, false>
+        inputId={props.id}
+        options={options}
+        value={value}
+        isLoading={isLoading}
+        onChange={onCreatableSelectChange}
+        components={{
+          Option,
+          SingleValue,
+        }}
+        filterOption={createFilter({
+          // search in the name and description
+          stringify: (option) => {
+            return `${option.data.name}${option.data.description}`
+          },
+        })}
+        noOptionsMessage={() => t('workspace.name.noOption', { type })}
+        formatCreateLabel={(inputValue) => t('workspace.name.createOption', { type, inputValue })}
+        onCreateOption={handleCreate}
+      />
+    </FormControl>
+  )
+}
+
+export const SchemaNameCreatableSelect = (props: WidgetProps) => {
+  const { data, isLoading } = useGetAllSchemas()
+  const options = useMemo<ResourceFamily[]>(() => {
+    if (!data) return []
+    if (!data.items) return []
+    const options = getSchemaFamilies(data.items)
+    return Object.values(options)
+  }, [data])
+
+  const createNewOption = (inputValue: string) => {
+    const schemaData = getNodePayload(DataHubNodeType.SCHEMA) as SchemaData
+    const newValue: ResourceFamily = {
+      name: inputValue,
+      versions: [1],
+      type: schemaData.type,
+      internalStatus: ResourceStatus.DRAFT,
+    }
+    return newValue
+  }
```

Comment:
suggestion: let's export this function so it does not get redefined one the component rerenders